h2/utilities.py

```diff
@@ -215,6 +225,24 @@ def _reject_uppercase_header_fields(headers, hdr_validation_flags):
         yield header
 
 
+def _reject_surrounding_whitespace(headers, hdr_validation_flags):
+    """
+    Raises a ProtocolError if any header name or value is surrounded by
+    whitespace characters.
+    """
+    for header in headers:
+        if header[0] and ((header[0][0] in _WHITESPACE) or
+                          (header[0][-1] in _WHITESPACE)):
+            raise ProtocolError(
+                "Received header name surrounded by whitespace %r" % header[0])
+        if header[1] and ((header[1][0] in _WHITESPACE) or
```

Comment:
Do you need to test the existence of the header fields? I suppose you're only testing whether they're empty, but empty header field names or values are a bad idea and so we shouldn't tolerate them at this time.